cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/assembly/KafkaRebalanceAssemblyOperator.java

```diff
@@ -322,19 +326,23 @@ private String rebalanceStateConditionType(KafkaRebalanceStatus status) {
                                    Set<Condition> unknownAndDeprecatedConditions) {
 
         log.info("{}: Rebalance action from state [{}]", reconciliation, currentState);
+        log.info("" + kafkaRebalance.getMetadata().getNamespace());
+        log.info("" + reconciliation.namespace());
 
         RebalanceOptions.RebalanceOptionsBuilder rebalanceOptionsBuilder = convertRebalanceSpecToRebalanceOptions(kafkaRebalance.getSpec());
 
         return computeNextStatus(reconciliation, host, apiClient, kafkaRebalance, currentState, rebalanceAnnotation, rebalanceOptionsBuilder)
-           .compose(desiredStatus -> {
+           .compose(desiredStatusandMap -> {
                // More events related to resource modification might be queued with a stale state. (potentially updated by the rebalance holding the lock)
                // Due to possible long rebalancing operations that take the lock for the entire period,
                // do a new get to retrieve the current resource state.
                return kafkaRebalanceOperator.getAsync(reconciliation.namespace(), reconciliation.name())
                             .compose(currentKafkaRebalance -> {
                                 if (currentKafkaRebalance != null) {
-                                    addWarningsToStatus(desiredStatus, unknownAndDeprecatedConditions);
-                                    return updateStatus(currentKafkaRebalance, desiredStatus, null)
+                                    addWarningsToStatus(desiredStatusandMap.getStatus(), unknownAndDeprecatedConditions);
+                                    return configMapOperator.reconcile(kafkaRebalance.getMetadata().getNamespace(),
+                                            kafkaRebalance.getMetadata().getName(), desiredStatusandMap.getLoadMap())
+                                            .compose(i -> updateStatus(currentKafkaRebalance, desiredStatusandMap.getStatus(), null))
```

Comment:
I pulled the branch and ran the `KafkaRebalanceAssemblyOperatorTest` class. This showed a load of null pointer exceptions centered on this line. Probably as a result of `desiredMapAndStatus` being null so calling `.getStatus()` causes the error. Are you sure that `computeNextStatus` never returns a `null` object?